### ISSUE TITLE
[Bug]: Table cuts off records from view on last page of multi-page query

### ISSUE THREAD

**Issue by leestreetnet** [appsmithorg/appsmith#35236]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I try to view a table that has multiple pages, the final page will not display all the records if the remaining records is equal or near the pageSize value of the table.

Example:

- A query has 63 records.
- A table has a pageSize of 16, with Default height as the property.
- There will be a total of 4 pages needed to view the records (63/16 = 3.9)
- Pages 1 - 3 display records 1-14 completely, with record 15 slightly cropped and record 16 not visible. The table has the ability for the user to scroll on these pages to see records 15 and 16 fully.
- The final page, 4 in this example, follows the same viewing pattern EXCEPT the scrolling functionality is no longer available which leaves record 15 slightly cropped and the final record (16) entirely inaccessible.

I marked this as 'Medium' but not displaying all records - when each are critically important to our operations - could be considered 'High' at least it is in our workflow.

[Ticket](https://internal.appsmith.com/app/oracle/ticket-details-662d74ffc3e4d34941e56ede?zid=2402)

### Steps To Reproduce

Using public sample below:

1. Preview pages 1-3 and observe all 16 records using the scrolling functionality.
2. Preview page 4 and notice only 14 records fully visible, 15 slightly cropped, and 16 inaccessible due to the scrolling functionality disabled.

### Public Sample App

https://actve.network/app/testing-table/page1-66a4049073fe0449d79b534e?environment=production

### Environment

Production

### Severity

Medium (Frustrating UX)

### Version

Self Hosted - v1.26

**Comment by Nikhil-Nandagopal** [comment]
@leestreetnet can you set the query to read the limit as the {{Table1.pageSize}} so that there is no scroll on the first page. This way your records become predictable and you can adjust the table height to perfectly fit the last record of the page

**Comment by leestreetnet** [comment]
@Nikhil-Nandagopal it is already currently set to {{Table1.pageSize}}. This is the query:

`SELECT id, name FROM public."users" 
LIMIT {{Table1.pageSize}}
OFFSET {{Table1.pageOffset}};`

I updated the Sample App to show the differences between a table with the same properties, one with data and one without. The Default row height is set to 'Default' for both tables, yet you can see the difference in row height between the table with data and the data without.

![image](https://github.com/user-attachments/assets/435f6f7d-6850-41a3-b43d-266319a89c4b)

**Comment by Nikhil-Nandagopal** [comment]
@leestreetnet I understand there is a difference in the heights of the table and that does look like a bug but I'm still unable to understand the situation where you are unable to see the last row of the 4th page. The app only has 2 pages of data and I can see both

**Comment by leestreetnet** [comment]
@Nikhil-Nandagopal I was using the Appsmith 'users' table for sample data and that likely is reset now that I think about it. So at the time I had added records so the total count was 64, but I see it is now back down to 23 records.

I changed the datasource to my own hosted database that has 64 records, I see now the final page does have the scrolling functionality now - was anything changed on your end? 

I also updated the sample app to show the difference in table row heights when 'Server Pagination' is turned ON vs OFF. The data is the same for each, but the height changes based on that setting.
